### PR TITLE
Feature/bookmark

### PR DESCRIPTION
**Pull request by williamchong007**

### CODE REVIEW

File: src/components/ContentCard.vue [hunk 1]
```diff
@@ -122,6 +140,9 @@ export default {
       return { backgroundImage: `url(${this.coverSrc})` };
     },
   },
+  methods: {
+    ...mapActions(['isInBookmark']),
```
Comment (nwingt) [inline src/components/ContentCard.vue]: Is used in the file?

File: src/components/ContentCardWrapper.vue [hunk 2]
```diff
@@ -8,6 +8,8 @@
     :description="internalDescription"
     :cover-src="internalCoverSrc"
     :like-count="internalLikeCount"
+    :is-bookmarked="getIsInBookmark(referrer)"
+    @bookmarkClick="onClickBookmark(referrer)"
```
Comment (nwingt) [inline src/components/ContentCardWrapper.vue]: Style guide says use `bookmark-click` in template

File: src/pages/index.vue [hunk 3]
```diff
@@ -106,10 +106,23 @@ export default {
...
+    ...mapActions(['refreshBookmarkList']),
+    async fetchSharedContent() {
+      try {
+        const promises = [];
+        if (!this.getUserBookmarks.length) promises.push(this.refreshBookmarkList());
+        await Promise.all(promises);
+      } catch (err) {
+        console.error(err); // eslint-disable-line no-console
+      }
+    },
```
Comment (nwingt) [inline src/pages/index.vue]: For prefetching?

File: src/store/modules/reader.js [hunk 4]
```diff
@@ -40,12 +44,25 @@ const mutations = {
...
   },
+  [READER_SET_BOOKMARK](state, bookmarks) {
+    state.bookmarks = bookmarks.reduce((acc, b) => {
+      acc[b] = true;
+      return acc;
+    }, {});
+  },
+  [READER_ADD_BOOKMARK](state, bookmark) {
+    Vue.set(state.bookmarks, bookmark, true);
+    state.bookmarks.delete(bookmark);
```
Comment (nwingt) [inline src/store/modules/reader.js]: What is this for?

File: src/pages/index.vue (hunk 3 above)
Comment (williamchong007) [reply src/pages/index.vue]: It depends if a content card needs to know if its bookmarked or not, in this PR's case the `Add bookmark` or `Remove bookmark` text needs this info across `/index`

**Pull request merged**